tests/test_fluentd.py

```diff
@@ -29,16 +29,14 @@ def test_fluentd(self):
         check.run()
         metrics = check.get_metrics()
         for m in metrics:
-            if m[0] == 'fluentd.forward.retry_count':
-                self.assertEquals(m[2], 0)
-            elif m[0] == 'fluentd.forward.buffer_queue_length':
-                self.assertEquals(m[2], 0)
-            elif m[0] == 'fluentd.forward.buffer_total_queued_size':
-                self.assertEquals(m[2], 0)
-            self.assertEquals(m[3]['type'], 'gauge')
+            if m[0] == 'fluentd.buffer_total_queued_size':
+                self.assertEquals(m[3]['type'], 'rate')
+            else:
+                self.assertEquals(m[3]['type'], 'gauge')
+            self.assertEquals(m[2], 0)
             self.assertEquals(m[3]['tags'], ['plugin_id:plg1'])
 
-        self.assertEquals(len(metrics), 3, metrics)
+        self.assertEquals(len(metrics), 2, metrics)
```

Comment:
you can call check.run() a second time. This way you would collect rate metrics as well.